pkg/fanal/analyzer/analyzer.go

```diff
@@ -214,7 +214,11 @@ func (r *AnalysisResult) Sort() {
 
 	// Misconfigurations
 	sort.Slice(r.Misconfigurations, func(i, j int) bool {
-		return r.Misconfigurations[i].FilePath < r.Misconfigurations[j].FilePath
+		if r.Misconfigurations[i].FileType != r.Misconfigurations[j].FileType {
+			return r.Misconfigurations[i].FileType < r.Misconfigurations[j].FileType
+		} else {
+			return r.Misconfigurations[i].FilePath < r.Misconfigurations[j].FilePath
+		}
```

Comment:
The change is nice and welcome but I'm curious why in this PR?

Reply:
The result may contain multiple misconfigurations for different scanners. For example, a YAML file can be scanned by YAML and CloudFormation scanners.  And to make the result deterministic, I will add sorting by type.